### PR TITLE
HBASE-24825 Add UncaughtExceptionHandler for NettyRpcConnection Relogin thread

### PR DESCRIPTION
**Pull request by nyl3532016**
We encountered a problem. The client access to the server kept reporting "Can not send request because relogin is in progress.", and there was no obvious exception log. Finally, it was found that an exception that could not be caught occurred during the execution of relogin(), causing the thread to exit silently, After this patch can get the exception below

`2020-08-06 14:59:51.664 WARN  org.apache.hadoop.hbase.ipc.NettyRpcConnection               - relogin failed
java.lang.NoSuchMethodError: org.apache.hadoop.security.UserGroupInformation.isLoginKerberosKeyBased()Z
        at org.apache.hadoop.hbase.ipc.RpcConnection.relogin(RpcConnection.java:180) ~[zjyprc-hadoop-flink1.9-xmpush-log-to-trace-span-xmpush-flink-talos-task-1.3-SNAPSHOT.jar:?]
        at org.apache.hadoop.hbase.ipc.NettyRpcConnection$1.run(NettyRpcConnection.java:162) ~[zjyprc-hadoop-flink1.9-xmpush-log-to-trace-span-xmpush-flink-talos-task-1.3-SNAPSHOT.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_202]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_202]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_202]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_202]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_202]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_202]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]`

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 42s |  master passed  |
| +1 :green_heart: |  compile  |   0m 25s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 32s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 47s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m  7s |  hbase-client in the patch passed.  |
|  |   |  23m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 19f5bbb56848 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c39cad220d |
| Default Java | 1.8.0_232 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/testReport/ |
| Max. process+thread count | 341 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  4s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 43s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 29s |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 21s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m  4s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   1m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  29m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux ccbaaa28df7a 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c39cad220d |
| Max. process+thread count | 94 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 57s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 16s |  master passed  |
| +1 :green_heart: |  compile  |   0m 30s |  master passed  |
| +1 :green_heart: |  shadedjars  |   7m 32s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 32s |  hbase-client in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 18s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 33s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 33s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m  2s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 28s |  hbase-client in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 18s |  hbase-client in the patch passed.  |
|  |   |  30m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 250214f0a86a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c39cad220d |
| Default Java | 2020-01-14 |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/testReport/ |
| Max. process+thread count | 277 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java [hunk 1]
```diff
@@ -178,7 +179,7 @@ private void scheduleRelogin(Throwable error) {
     RELOGIN_EXECUTOR.schedule(() -> {
       try {
         provider.relogin();
-      } catch (IOException e) {
+      } catch (Throwable e) {
```
Comment (wchevreuil) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java:182]: Yeah, I think it makes sense. IIRC, netty swallows the throwable silently, so it's difficult to figure out what's wrong.

**Comment by Apache9** [comment]
Mind exlaining the reason a bit?

**Comment by nyl3532016** [comment]
> LGTM. Would other parts of the connection attempt catching only IOException also benefit from this change?
I think other parts catching only IOException is just fine, the uncatched exception will throw upward

File: hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java (hunk 1 above)
Comment (nyl3532016) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java:182]: I think other parts catching only IOException is just fine, the uncatched exception will throw upward

**Comment by nyl3532016** [comment]
> Mind exlaining the reason a bit?
Yes, I add a description

**Comment by Apache9** [comment]
I think this is a Xiaomi specific problem? Seems you have compiled with a customized hadoop library but then someone introduces a official library?

**Comment by infraio** [comment]
> I think this is a Xiaomi specific problem? Seems you have compiled with a customized hadoop library but then someone introduces a official library?

Yes. But we still have chance to meet other incompatibility problem or other RuntimeException?

**Comment by Apache9** [comment]
Add an uncaught exception handler to the Thread created by the thread factory? So we could log an error for this 'unexpected' exception.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 28s |  master passed  |
| +1 :green_heart: |  compile  |   0m 23s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 36s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 20s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 13s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 20s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m  1s |  hbase-client in the patch passed.  |
|  |   |  23m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 2fac5f0eee75 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 485e0d2fa4 |
| Default Java | 1.8.0_232 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/testReport/ |
| Max. process+thread count | 342 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  8s |  master passed  |
| +1 :green_heart: |  compile  |   0m 30s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m  0s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 26s |  hbase-client in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 15s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 41s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 26s |  hbase-client in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 11s |  hbase-client in the patch passed.  |
|  |   |  24m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 1247f21a165a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 485e0d2fa4 |
| Default Java | 2020-01-14 |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/testReport/ |
| Max. process+thread count | 274 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  2s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 28s |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m  1s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 41s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  12m 18s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   1m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  32m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux 353180464bef 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 485e0d2fa4 |
| Max. process+thread count | 84 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by nyl3532016** [comment]
@virajjasani when submitting this patch, I find maybe  the default UncaughtExceptionHandler (Threads.LOGGING_EXCEPTION_HANDLER) will unset in  ThreadFactoryBuilder.dobuild after HBASE-24750, It will be an Inconsistent ?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 44s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 39s |  master passed  |
| +1 :green_heart: |  compile  |   0m 32s |  master passed  |
| +1 :green_heart: |  shadedjars  |   7m 10s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 30s |  hbase-client in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 16s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 30s |  hbase-client in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 24s |  hbase-client in the patch passed.  |
|  |   |  30m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 2b43d99f08de 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3edfa1025c |
| Default Java | 2020-01-14 |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/testReport/ |
| Max. process+thread count | 262 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  5s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  1s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 28s |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m  6s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 40s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  12m 13s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   1m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  32m  4s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux 0728b4e4a94c 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3edfa1025c |
| Max. process+thread count | 84 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  42m 51s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 46s |  master passed  |
| +1 :green_heart: |  compile  |   0m 25s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 43s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 15s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m  9s |  hbase-client in the patch passed.  |
|  |   |  66m 10s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2206 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 22b43f1714bd 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3edfa1025c |
| Default Java | 1.8.0_232 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/testReport/ |
| Max. process+thread count | 340 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2206/3/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
> @virajjasani when submitting this patch, I find maybe the default UncaughtExceptionHandler (Threads.LOGGING_EXCEPTION_HANDLER) will unset in ThreadFactoryBuilder.dobuild after HBASE-24750, It will be an Inconsistent ?

We are not planning to remove `Threads.LOGGING_EXCEPTION_HANDLER`, it will stay. I hope the expectation for this patch is to handle the exception and just log it, and to not propagate it back to caller. If so, using this default UncaughtExceptionHandler from Threads class is fine.

**Comment by virajjasani** [comment]
Do you think using default UncaughtExceptionHandler will provide better warn logs if something goes wrong with relogin()? Maybe try creating a custom UncaughtExceptionHandler with `NettyRpcConnection` specific log message so that it is easier to figure out the issue runtime?

**Comment by nyl3532016** [comment]
> Do you think using default UncaughtExceptionHandler will provide better warn logs if something goes wrong with relogin()? Maybe try creating a custom UncaughtExceptionHandler with `NettyRpcConnection` specific log message so that it is easier to figure out the issue runtime?

I think LOGGING_EXCEPTION_HANDLER is find, It will log Thread name

**Comment by nyl3532016** [comment]
[#2231](https://github.com/apache/hbase/pull/2231) has resolved this issue, close this PR

**Pull request closed without merging**